### PR TITLE
[PW-6181] Add workflow for nuget package

### PR DESCRIPTION
**Pull request by jillingk**
**Description**
Provide workflow to upload packages to Nuget. We just need to enable a github secret for the nuget api key.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,18 +1,10 @@
-$script = <<-SCRIPT
-sudo rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm
-echo "Run update"
-sudo yum update
-echo "Install dotnet core"
-sudo yum -y install dotnet-sdk-2.2
-SCRIPT
 Vagrant.configure("2") do |config|
-  config.vm.box = "centos/7"
+  config.vm.box = "jeffnoxon/ubuntu-20.04-arm64"
```
Comment (wboereboom) [inline Vagrantfile:2]: why use this box rather than the official ubuntu one?

Comment (wboereboom) [reply Vagrantfile:2]: https://app.vagrantup.com/generic/boxes/ubuntu2004
https://app.vagrantup.com/generic/boxes/ubuntu2010

Comment (jillingk) [reply Vagrantfile:2]: Honestly never tried those, used the jeffnoxon bc it's lighter

Comment (wboereboom) [reply Vagrantfile:2]: Let's try and use the official one wherever possible

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Run test suite netcoreapp2.1
+        run: dotnet test --no-build -c Debug -f netcoreapp2.1 Adyen.Test/Adyen.Test.csproj
+      - name: Run test suite net6.0
+        run: dotnet test --no-build -c Debug -f net6.0 Adyen.Test/Adyen.Test.csproj
+      # Pack Release
+      - name: Pack
+        run: dotnet pack Adyen --no-build --configuration Release --output nuget
+      # Publish Package
+      - name: Publish Nuget
+        run: dotnet nuget push nuget/*.nupkg --api-key ${{ secrets.NUGET_KEY }} --source "nuget.org"
```
Comment (AlexandrosMor) [inline .github/workflows/publish.yml:33]:
```suggestion
        run: dotnet nuget push nuget/*.nupkg --api-key ${{ secrets. ADYEN_NUGET }} --source "nuget.org"
```

**Review by michaelpaul — APPROVED** [review]

**Pull request merged**